cachito/workers/tasks/utils.py

```diff
@@ -203,3 +205,69 @@ def _get_request_or_fail(request_id: int, connect_error_msg: str, status_error_m
         raise CachitoError(msg)
 
     return rv.json()
+
+
+class PackagesData:
+    """A collection of resolved packages."""
+
+    def __init__(self) -> None:
+        self._index = set()
+        self._packages = []
+
+    def add_package(self, pkg_info: Dict[str, str], path: str, deps: List[Dict[str, Any]]) -> None:
+        """Add a package with deps.
+
+        :param dict[str, str] pkg_info: a mapping containing a package information.
+            It must have ``name``, ``type`` and ``version`` key/value pairs.
+        :param str path: the path where the package is retreived. Consult with the
+            ``fetch_*_source`` for the defailed information about a package's path.
+        :param deps: a list of depencencies the package has.
+        :type deps: list[dict[str, any]]
+        :raises CachitoError: if there is a package with same name, type and version
+            has been added already.
+        """
+        key = (pkg_info["name"], pkg_info["type"], pkg_info["version"])
+        if key in self._index:
+            raise CachitoError(f"Duplicate package: {pkg_info!r}")
+        self._index.add(key)
+        self._packages.append(
+            {
+                "name": pkg_info["name"],
+                "type": pkg_info["type"],
+                "version": pkg_info["version"],
+                "path": path,
```

Comment:
Currently, the database will only store the `path` if `path != "."`. Doing the same thing for the JSON file would make it slightly easier to use it in the API later.